commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/unicycle/UnicycleDrive.scala

```diff
@@ -2,47 +2,55 @@ package com.lynbrookrobotics.potassium.commons.drivetrain.unicycle
 
 import com.lynbrookrobotics.potassium.Signal
 import com.lynbrookrobotics.potassium.commons.drivetrain._
-import com.lynbrookrobotics.potassium.commons.drivetrain.purePursuit.{PurePursuitControllers, PurePursuitTasks}
+import com.lynbrookrobotics.potassium.commons.drivetrain.unicycle.control.purePursuit.{PurePursuitControllers, PurePursuitTasks}
+import com.lynbrookrobotics.potassium.commons.drivetrain.unicycle.control.{UnicycleCoreControllers, UnicycleCoreTasks, UnicycleMotionProfileControllers}
 import com.lynbrookrobotics.potassium.streams.Stream
 import squants.Percent
 
 /**
   * A drivetrain that has forward-backward and turning control in the unicycle model
   */
-trait UnicycleDrive extends Drive { self =>
+trait UnicycleDrive extends Drive {
+  self =>
   type Hardware <: UnicycleHardware
   type Properties <: UnicycleProperties
 
   /**
     * Converts a unicycle signal value to the parent's signal type
+    *
     * @param uni the unicycle value to convert
     * @return the parent's equivalent signal
     */
-  protected def convertUnicycleToDrive(uni: UnicycleSignal): DriveSignal
-  
-  object UnicycleControllers extends PurePursuitControllers with UnicycleMotionProfileControllers with UnicycleCoreControllers{
+  protected def unicycleToOpenLoopSignal(uni: UnicycleSignal): OpenLoopSignal
+
+  object UnicycleControllers extends PurePursuitControllers with UnicycleMotionProfileControllers with UnicycleCoreControllers {
     type DriveSignal = self.DriveSignal
+    type OpenLoopSignal = self.OpenLoopSignal
     type DrivetrainHardware = self.Hardware
     type DrivetrainProperties = self.Properties
 
+    override def openLoopToDriveSignal(openLoop: OpenLoopSignal): DriveSignal =
+      self.openLoopToDriveSignal(openLoop)
+
     /**
-      * Uses the parent's open loop control for the equivalent drive signal for the unicycle signal
+      * Uses the childs's open loop control for the equivalent drive signal for the unicycle signal
+      *
       * @param unicycle the unicycle signal to drive with
-      * @return a signal controlled with open-loop on the parent
+      * @return a signal controlled with open-loop on the child
       */
-    def parentOpenLoop(unicycle: Stream[UnicycleSignal]): Stream[DriveSignal] = {
-      unicycle.map(convertUnicycleToDrive)
-    }
+    def childOpenLoop(unicycle: Stream[UnicycleSignal]): Stream[OpenLoopSignal] =
```

Comment:
I can't find any places where we don't follow this with a `map(openLoopToDriveSignal)`. I think we can change the return type to `Stream[DriveSignal]` and include the `openLoopToDriveSignal` transformation in the body itself.